index.js

```diff
@@ -189,7 +199,113 @@ class Squeakquel extends Datastore {
         });
 
         // @TODO Indexes and Range Keys
-        return this.client.define(tableName, tableFields, tableOptions);
+        const tableObj = this.client.define(tableName, tableFields, tableOptions);
+
+        try {
+            if (modelName === 'build' && this.captureBuildMetrics === 'true') {
+                tableObj.afterCreate('afterCreateHook', (instance, options) => {
+                    this.upsertBuildReportTable(instance, options, this.prefix);
+                });
+                tableObj.afterUpdate('afterUpdateHook', (instance, options) => {
+                    this.upsertBuildReportTable(instance, options, this.prefix);
+                });
+            }
+        } catch (err) {
+            logger.error(`Error upserting build metrics table: ${err}`);
+        }
+
+        return tableObj;
+    }
+
+    /**
+     * upsertBuildReportTable - any change in build status, upsert into flattened build metrics table
+     * @param  {Object} instance build table object
+     * @param  {Object} options  timestamps and indexes
+     * @param  {String} prefix   schema prefix - beta or blank
+     */
+    upsertBuildReportTable(instance, options, prefix) {
+        logger.info(`Started capturing build metrics for build id: ${instance.id}`);
+        if (instance._previousDataValues.status !== instance.dataValues.status) {
+            this.client.query(`INSERT INTO "${prefix}buildReports" ( ` +
+              ' "buildId", "jobId", "parentBuildId", "number", "created", ' +
```

Comment:
1. triggers n trigger functions, but it doesn't go with sequelize.
2. sequelize (afterSync), it will also not work as we dont have a way to create trigger if not exists
3. thought of models, but this is a separate flow though
lets group n decide on this

Reply:
Yeah, let's schedule some time tomorrow. I think what Dao means is to convert this query into sequelize function

Reply:
this PR is updated to only create index